sqlglot/parser.py

```diff
@@ -1824,16 +1824,22 @@ def _parse_property(self) -> t.Optional[exp.Expression]:
             return self.expression(exp.SqlSecurityProperty, definer=self._match_text_seq("DEFINER"))
 
         index = self._index
-        key = self._parse_column()
+        key = self._parse_var(any_token=True) if parse_as_var else self._parse_column()
 
         if not self._match(TokenType.EQ):
             self._retreat(index)
             return self._parse_sequence_properties()
 
+        value = self._parse_bitwise() or self._parse_var(any_token=True)
```

Comment:
why do we need this fallback _parse_var?

Reply:
This conjuction pre-existed and by removing the fallback there's a T-SQL test breaking when trying to parse the option `PAD_INDEX = ON` (`parse_bitwise` failed because of reserve keyword afaict) so I transferred it as is